src/basic/escape.h

```diff
@@ -34,8 +34,9 @@ typedef enum UnescapeFlags {
 } UnescapeFlags;
 
 typedef enum EscapeStyle {
-        ESCAPE_BACKSLASH = 1,
-        ESCAPE_POSIX     = 2,
+        ESCAPE_BACKSLASH         = 1,
+        ESCAPE_BACKSLASH_ONELINE = 2,
+        ESCAPE_POSIX             = 3,
```

Comment:
any chance you can add very brief side comments here, clarifying in one well-chosen example each what each of these choices do? the more options this becomes the worse it is to follow what this does...